### ISSUE TITLE
Use undocumented native endpoints to leave removal reason comments/modmails

### ISSUE THREAD

**Issue by eritbh** [toolbox-team/reddit-moderator-toolbox#582]
this issue is written in a really roundabout way because i'm tired and have no energy sorry. but yes this is about "remove as a subreddit"

As pointed out on Discord [here](https://discord.com/channels/535490452066009090/540164483025469490/1009568435803533393), there's an unoffical/undocumented native endpoint for submitting removal reasons. It allows you to send the reason as a comment under a built-in subreddit shared account (this is new and recent). I would guess it will also be used for other things in the future, e.g. to show removal reasons left on users in mod notes or something like that.

The (undocumented) endpoints are `/api/v1/modactions/removal_comment_message` and `/api/v1/modactions/removal_link_message` which take a body like

```jsonc
{
  "item_id": [
    "t1_ikodx0k"
  ],
  "message": "comment goes here",
  "title": "rule title goes here?",
  "type": "public_as_subreddit", // or `public`, `private`, `private_exposed`
  "lock_comment": false
}
```

The endpoints don't appear to require any hard link to the native removal reasons data (the `"title"` field is... something? but it doesn't appear to be reflected anywhere), so we should theoretically be able to modify the current Toolbox removal reasons module to use these endpoints to submit comments/modmails when leaving reasons with arbitrary text. We can also use this to support sending removal reason comments as the subreddit.

**Comment by creesch** [comment]
I did some testing and this should work. The only catch is that if the item isn't actually removed, the API will return a nonsense error about the ID not being correct.

**Comment by creesch** [comment]
Some additional thoughts: 

- Do we want to replace the API calls toolbox currently uses in favor of these calls? Or do we want to provide the undocumented native API endpoint as an option with a big fat disclaimer? 
- Do we want to build in some sort of fallback to the proven API calls in case the native ones decide to crap out or change payload requirements for whatever reason?

**Comment by eritbh** [comment]
I think I'm in favor of replacing the existing calls with these but using our existing calls as fallbacks. As far as I can tell, the removal reasons API this is related to has been pretty stable, so I'm not terribly worried about things drastically changing. Ideally we can eventually convince Reddit to document the `modactions` APIs and then we can remove the fallbacks, but until then I think that makes sense to keep a fallback for what we can just in case.

**Comment by creesch** [comment]
Seems like a sensible approach. I am not entirely sure we can avoid using both our current endpoints and the new ones though, unless we want to take away options (and you know how well that can go over at times. 

Frankly I'd be in favor of no longer offering the option to send as a personal PM. But that does leave a few other things. 

So I am thinking that we might add a third radio button stating `Reply through subreddit account with a comment to the item that is removed.` and only use the reddit api for that to begin with. So something like this: 

![image](https://user-images.githubusercontent.com/2502190/185447593-f1ba1d69-0575-4bc6-9166-375a02003a73.png)

We can then trial this a bit and see how stable it is while leaving everything else as is. 

Alternatively, we could do something like this: 

![image](https://user-images.githubusercontent.com/2502190/185447862-e338a1b9-c1db-47e7-99ea-6c708c90536d.png)

And replace the entire comment action with the reddit rr api call. I am however not sure if reddit removal reasons do sticky (by default) and if they don't if the removal reason API returns the ID of the posted comment.

**Comment by creesch** [comment]
I just checked by monitoring network in devtools when adding a removal reason on new reddit. It appears you get the ID back of the posted removal reason. However it seems like reddit removal reasons do sticky automatically which you can't undo as it is a different account...

**Comment by eritbh** [comment]
Fixed in #583
